### PR TITLE
Include user validation as a plugin

### PR DESCRIPTION
**Pull request by steve-gregory**
This is a "first stab" attempt at validating AtmosphereUser in a pluggable way. Half of the work for VALIDATION_PLUGINS was swept into changes related to AllocationSource, this will bring in the other half of that work.

### CODE REVIEW

**Comment by julianpistorius** [comment]
👍

**Comment by steve-gregory** [comment]
This PR is ready for review 👓  After which point we will merge into `jets-qq-release`

**Comment by julianpistorius** [comment]
Looked good to me.

File: variables.ini.dist [hunk 1]
```diff
@@ -107,6 +107,7 @@ DEFAULT_PASSWORD_UPDATE = ; 'keystone_password_update', 'openstack_sdk_password_
 INSTANCE_HOSTNAMING_DOMAIN = ; iplantcollaborative.org
 INSTANCE_HOSTNAMING_FORMAT = ; vm%(three)s-%(four)s.%(domain)s
 DEPLOYMENT_KEYPAIR_NAME = ; keypair name inject by atmo-ansible during instance deploy
+VALIDATION_PLUGINS = ; ["atmosphere.plugins.auth.validation.AlwaysAllow"]
```
Comment (lenards) [inline variables.ini.dist]: I think this means we need to updated `..../build_env/variables.yml` I think.

**Comment by lenards** [comment]
👍

**Comment by lenards** [comment]
Just curious about the variable update action.

**Comment by steve-gregory** [comment]
:bullettrain_front: :shipit:

**Pull request merged**